### DISCUSSION TITLE
Reverse Proxy Casa OS terminal issues

### DISCUSSION THREAD

**Discussion by NickTh3M4l4chi** [IceWhaleTech/CasaOS#338] [Q&A]
I am attempting to use a reverse proxy on casa OS doing so via my old NAS.(Casa OS on my Pi, NAS is another machine altogether)

For whatever reason I can NOT get docker or the casa ssh terminal to do anything while connected via reverse proxy. It just shows a blank terminal like so.
![image](https://user-images.githubusercontent.com/108558258/177373932-c87dab64-b2fe-4b78-9455-3cde4a075c48.png)

If I connect via the local IP and port it all works fine.

I am using reverse proxy to take http and make it https for outside connections for when I am away from my home network.

I set the headers for the reverse proxy to be:
![image](https://user-images.githubusercontent.com/108558258/177374523-3a66ae67-99a3-405e-b4fe-78a49dd7aaa7.png)

do I need to add more? and if so what?

**Comment by AwokenPineal** [comment]
I don't have a solution just commenting to say that I am also experiencing this same problem. I've also noticed it stops the system hardware gauges from updating in real time as well.

**Reply by NickTh3M4l4chi** [reply]
Sorry they had fixed it what was causing it in an update for me. I can do everything using synology reverse proxy with the same settings.

**Reply by leesonaa** [reply]
you can try  gdy666/lucky

**Reply by robjo82** [reply]
Hello, I have the same problem actually, the same black screen and same captors freezing. I look for a solution ;)

**Comment by bcostantino** [comment]
In case anyone is still facing this issue, and you're seeing errors similar to below in the web console, it's likely caused by a failure to proxy websocket connections.

![image](https://github.com/user-attachments/assets/e4100b81-aee0-4d16-b35f-644641ed86c4)

I've only been able to try this with Apache reverse proxy, but adding the following lines to the virtualhost resolved the issue for me (see https://github.com/dani-garcia/vaultwarden/discussions/3598 for original solution). Ensure that mod_rewrite is enabled and that the rewrite rule hostname and port are pointing at casaos web server.

```
RewriteEngine on
RewriteCond %{HTTP:Upgrade} websocket [NC]
RewriteCond %{HTTP:Connection} upgrade [NC]
RewriteRule ^/?(.*) "ws://localhost:88/$1" [P,L]
```

**Comment by Rizwan-Hasan** [comment]
This worked for me

```
location / {
    # redirect all HTTP traffic to localhost:8080
    proxy_pass http://localhost:8080;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

    # WebSocket support
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
}
```
